libs/community/langchain_community/document_loaders/pebblo.py

```diff
@@ -193,10 +214,17 @@ def _send_loader_doc(self, loading_end: bool = False) -> None:
             logger.warning("Unable to reach pebblo server.")
         except Exception as e:
             logger.warning("An Exception caught in _send_loader_doc: %s", e)
+        finally:
+            classified_docs=json.loads(pebblo_resp.text).get('docs', None)
+            if not isinstance(classified_docs, list):
+                classified_docs = None
 
         if self.api_key:
             try:
-                payload['docs'] = json.loads(pebblo_resp.text)['docs']
+                if not classified_docs:
+                    logger.warning("No classified docs to send to Daxa cloud.")
+                    return
```

Comment:
This scenario would occur in case if Pebblo Server is down or it has some issues and it sends empty response.
So handling it this way might be misleading.
What if we move whole api_key check within try block above? that way, if something goes wrong on the `/v1/loader/doc` API then we already have relevant exception messages there in exception handling.